pydantic/validators.py

```diff
@@ -126,7 +126,7 @@ def int_validator(v: Any) -> int:
 
     try:
         return int(v)
-    except (TypeError, ValueError):
+    except (TypeError, ValueError, OverflowError):
         raise errors.IntegerError()
```

Comment:
```suggestion
    except (TypeError, ValueError):
        raise errors.IntegerError()
    except OverflowError:
        raise errors.IntegerOverflowError()
```

I think this is meant.

Reply:
Then, like 144

```diff
-        return float(v)
+        rv = float(v)
+        if rv in (float("inf"), float("-inf")):
+            raise errors.FloatOverflowError()
+        if rv != rv:
+            raise errors.FloatNanError()
+        return rv
```

P.S. the weird line of code from https://stackoverflow.com/a/44154660/705086

Reply:
floats are a separate case. I would say that `float('inf')` is a valid float, arguably so is `float('nan')`.